### PR TITLE
fix(explorer): include companions in the duplicate result_id identity

### PR DESCRIPTION
**Pull request by joeharris76**
Addresses Codex **P1** on #1516 (already merged, so this is a follow-up). The finding is correct and it is a real hole in the guard I just added.

## The bug

The identical-content skip hashed **only the primary bundle**. But companions publish as `{result_id}.plans.json`, keyed by the same id. Two byte-identical primaries whose sidecars differ — one carrying plans, the other not — digested the same, so the second was treated as a redundant copy and skipped, **silently dropping its sidecar**.

That is the same class of evidence loss the guard was added to stop, one file over.

## Fix

Digest the primary plus every entry in `COMPANION_SUFFIXES`, with **absence itself part of the identity** so a bundle with no plans cannot match one that has them.

Companions are read raw, deliberately: this digest is an identity check and never a published artifact, so it needs no sanitizing — and reading raw means a companion differing only in private, later-redacted content still marks the two bundles distinct, which fails safe.

## Verification

- **Falsified** by digesting the primary only: the new test fails.
- `tests/unit/scripts/explorer_pipeline`: **324 passed**
- real 207-bundle corpus still builds (207 processed, 207 published)

## Identity findings on this PR: refuted

Codex also flagged the commit as authored by `Codex <codex@openai.com>`. That is incorrect — every commit is:

```
author=Joe Harris <joeharris76@gmail.com> | committer=Joe Harris <joeharris76@gmail.com>
```

Same for #1517, #1518 and #1521. On #1521 it cited commit `4fbd4b77a101`, which is not on that branch at all.

### CODE REVIEW

**Comment by joeharris76** [comment]
**P2 "Stream companion data into the digest" — correct, and it was a real regression. Fixed in `c86a1a1`.**

You're right that `read_bytes()` bypasses the `APPLIED_COMPANION_MAX_BYTES` guard in `_applied_receipt`, which deliberately stats the file and returns a truncation marker without reading it. My whole-file read could therefore exhaust memory on a corpus that previously built fine.

Companions are now hashed incrementally in bounded 1 MiB chunks, so memory stays flat regardless of file size.

Chunking is only correct if it covers the *whole* file, so the test flips a byte past the first chunk boundary — with a single read that is indistinguishable from the shorter file. Falsified by stopping after one chunk. 325 passed; the real 207-bundle corpus still builds.

**P1 identity finding — refuted.** Every commit on this branch is `author=Joe Harris <joeharris76@gmail.com> | committer=Joe Harris <joeharris76@gmail.com>`. Same on #1516, #1517, #1518 and #1521; on #1521 the cited commit `4fbd4b77a101` is not on that branch at all.

File: _project/scripts/explorer_pipeline/pipeline.py [hunk 1]
```diff
@@ -131,6 +131,36 @@ def _promote_staged_output(staged_dir: Path, output_dir: Path) -> None:
...
+    differing only in private, later-redacted content still marks the two
+    bundles distinct, which fails safe.
+    """
+    hasher = hashlib.sha256()
+    hasher.update(public_raw)
+    for suffix in COMPANION_SUFFIXES:
+        companion = bundle_path.with_name(f"{bundle_path.stem}{suffix}")
+        hasher.update(suffix.encode())
+        try:
+            hasher.update(hashlib.sha256(companion.read_bytes()).hexdigest().encode())
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply _project/scripts/explorer_pipeline/pipeline.py]: Already fixed on `develop` by #1532 (`c07d686c0c`) — no change needed.

`_publication_digest` reads companions in 1 MiB chunks through `iter(handle.read, b"")` instead of `read_bytes()`, so memory stays flat regardless of companion size and the `APPLIED_COMPANION_MAX_BYTES` guard is no longer bypassed. Resolving.

File: _project/scripts/explorer_pipeline/pipeline.py [hunk 2]
```diff
@@ -131,6 +131,36 @@ def _promote_staged_output(staged_dir: Path, output_dir: Path) -> None:
             logger.warning("Could not remove legacy artifact %s", output_dir / legacy_file)
 
 
+def _publication_digest(bundle_path: Path, public_raw: bytes) -> str:
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply _project/scripts/explorer_pipeline/pipeline.py:139]: Valid finding, but not actionable in a follow-up PR — flagging for a maintainer decision.

The commit is already merged, so re-authoring it means rewriting published history on `develop`: a force-push and invalidated SHAs for anything branched from it. That is a maintainer call, not something a review sweep should do unilaterally.

Worth noting this is one of **15** identity threads in the current merged-PR window, up from one in the previous sweep. The per-commit remedy is the wrong layer at that volume — something upstream is routinely authoring as `Codex <codex@openai.com>`, and the durable fix is to stop it at the source rather than re-author after the fact. Resolving to keep the queue readable; the underlying question stands on its own.
